cvpy/annotation/base/Project.py

```diff
@@ -0,0 +1,129 @@
+from typing import List
+
+from swat.cas import CAS, CASTable
+
+from cvpy.annotation.base.AnnotationLabel import AnnotationLabel
+from cvpy.annotation.base.AnnotationType import AnnotationType
+from cvpy.annotation.base.Credentials import Credentials
+
+
+class Project(object):
+    '''
+    This is a base class for a project in an annotation tool. It has several abstract methods that must be
+    implemented by a subclass.
+    '''
+
+    def __init__(self, cas_session: CAS = None, url: str = None, credentials: Credentials = None,
+                 project_name: str = None, annotation_type: AnnotationType = None, lables: List[AnnotationLabel] = None,
+                 project_id: str = None) -> None:
+        '''
+        Constructor for Project class
+        :param cas_session: the CAS session for this project
+        :param url: the url of the CVAT server for calling REST APIs
+        :param credentials: the login credentials to connect to the CVAT server
+        :param project_name: name of the project
+        :param annotation_type: the type of the annotation project
+        :param lables: a list of dicts
+        :param project_id:
+        '''
+        self._cas_session = cas_session
+        self._url = url
+        self._credentials = credentials
+        self._project_name = project_name
+        self._annotation_type = annotation_type
+        self._labels = lables
+        self._project_id = project_id
+
+    @property
+    def cas_session(self) -> CAS:
+        return self._cas_session
+
+    @cas_session.setter
+    def cas_session(self, cas_session) -> None:
+        self._cas_session = cas_session
+
+    @property
+    def url(self) -> str:
+        return self._url
+
+    @url.setter
+    def url(self, url: str) -> None:
+        self._url = url
+
+    @property
+    def credentials(self) -> Credentials:
+        return self._credentials
+
+    @credentials.setter
+    def credentials(self, credentials: Credentials) -> None:
+        self._credentials = credentials
+
+    @property
+    def project_name(self):
+        return self._project_name
+
+    @project_name.setter
+    def project_name(self, project_name: str) -> None:
+        self._project_name = project_name
+
+    @property
+    def annotation_type(self):
+        return self._annotation_type
+
+    @annotation_type.setter
+    def annotation_type(self, annotation_type: AnnotationType) -> None:
+        self._annotation_type = annotation_type
+
+    @property
+    def labels(self) -> List[AnnotationLabel]:
+        return self._labels
```

Comment:
I'm not entirely familiar with the decorators here, but it looks like you have already implemented setters and getters. Shruti had a story to design these implementations. Can you check with her to see if anything is missing? Make sure you're on the same page, and we can delete any future stories that Shruti has created related to this.

Reply:
I am checking with Shruti.

Reply:
Shruti pointed out a missing function for getting the tasks. I added that function and a supporting Task class. So Shruti can now delete the other story. I will let her know.